### ISSUE TITLE
swigpyrun missing some MPI location variable

### ISSUE THREAD

**Issue by VictorEijkhout** [trilinos/Trilinos#3215]
Which variable am I missing?

```
[ 97%] Generating swigpyrun.h
cd /tmp/trilinos-build/packages/PyTrilinos/src && /opt/apps/swig/3.0/bin/swig -python -external-runtime
cd /tmp/trilinos-build && /opt/apps/cmake/3.10.2/bin/cmake -E cmake_depends "Unix Makefiles" /admin/rpms/BUILD/trilinos-git /admin/rpms/BUILD/trilinos-git/packages/PyTrilinos/src /tmp/trilinos-bu
ild /tmp/trilinos-build/packages/PyTrilinos/src /tmp/trilinos-build/packages/PyTrilinos/src/CMakeFiles/pytrilinos.dir/DependInfo.cmake --color=
Traceback (most recent call last):
  File "/tmp/trilinos-build/packages/PyTrilinos/src/gen_teuchos_rcp.py", line 647, in <module>
    main()
  File "/tmp/trilinos-build/packages/PyTrilinos/src/gen_teuchos_rcp.py", line 546, in main
    print("#define MPI_VERSION %s" % get_mpi_version())
  File "/tmp/trilinos-build/packages/PyTrilinos/src/gen_teuchos_rcp.py", line 47, in get_mpi_version
    for line in open(header, 'r').readlines():
IOError: [Errno 2] No such file or directory: '/include/mpi.h'
make[2]: *** [packages/PyTrilinos/src/Teuchos.RCP.i] Error 1
make[2]: *** Deleting file `packages/PyTrilinos/src/Teuchos.RCP.i'
make[2]: Leaving directory `/tmp/trilinos-build'
make[1]: *** [packages/PyTrilinos/src/CMakeFiles/PyTrilinos_Teuchos_RCP.dir/all] Error 2
```

**Comment by wfspotz** [comment]
It would appear my `gen_teuchos_rcp.py` script expects environment variable `MPI_BASE_DIR` to be set. The workaround would be to set it. How do you currently set MPI when you call `cmake`?

**Comment by VictorEijkhout** [comment]
Is that a new variable? I used to specify the mpicc & such compilers.

Ok, with that variable:

```
/admin/rpms/BUILD/trilinos-git/packages/teuchos/comm/src/Teuchos_DefaultComm.hpp:291: Error: CPP #error ""Sorry, you need an MPI implementation that supports at least MPI 2.0 in order to build th
is code.  MPI 2.0 came out in 1997.  I wrote this comment in 2017.  If you really _really_ want MPI 1.x support, please file a GitHub issue for this feature request at github.com/trilinos/trilino
s/issues with an expression of its priority and we will get to it as soon as we can."". Use the -cpperraswarn option to continue swig processing.
/admin/rpms/BUILD/trilinos-git/packages/teuchos/comm/src/Teuchos_DefaultComm.hpp:385: Error: CPP #error ""Sorry, you need an MPI implementation that supports at least MPI 2.0 in order to build th
is code.  MPI 2.0 came out in 1997.  I wrote this comment in 2017.  If you really _really_ want MPI 1.x support, please file a GitHub issue for this feature request at github.com/trilinos/trilino
s/issues with an expression of its priority and we will get to it as soon as we can."". Use the -cpperraswarn option to continue swig processing.
make[2]: *** [packages/PyTrilinos/src/Teuchos.RCPPYTHON_wrap.cpp] Error 1
make[2]: Leaving directory `/tmp/trilinos-build'
make[1]: *** [packages/PyTrilinos/src/CMakeFiles/PyTrilinos_Teuchos_RCP.dir/all] Error 2
```

**Comment by wfspotz** [comment]
Hmmm. I'm assuming you have used this MPI before and it has not resulted in Teuchos complaining that it is too old. I may be parsing the `mpi.h` file incorrectly. Can you attach your `$MPI_BASE_DIR/include/mpi.h` ?

**Comment by wfspotz** [comment]
OK, I see the problem. In the source directory

    Trilinos/packages/PyTrilinos/src/gen_teuchos_rcp.py.in

on line 49, we need to change

    version = line.split(" ")[2]

to

    version = line.split()[2]

You could do this directly in your copy as a temporary workaround while I update it in the repository. As a secondary fix, I would like the logic to work in your original case. It sounds like you called `cmake` with the following options:

    -D TPL_ENABLE_MPI:BOOL=ON \
    -D MPI_CXX_COMPILER:FILEPATH=<file-path-to-mpi-cxx-compiler> \
    -D MPI_C_COMPILER:FILEPATH=<file-path-to-mpi-c-compiler>

Do I have that right? Do you need to do anything else?

**Comment by wfspotz** [comment]
Fixed with pull request #3569
